src/cryptonote_core/blockchain.cpp

```diff
@@ -3798,17 +3818,14 @@ bool Blockchain::check_block_timestamp(const block& b, uint64_t& median_ts) cons
     return true;
   }
 
-  std::vector<uint64_t> timestamps;
-
-  // need most recent 60 blocks, get index of first of those
-  size_t offset = h - BLOCKCHAIN_TIMESTAMP_CHECK_WINDOW;
-  timestamps.reserve(h - offset);
-  for(;offset < h; ++offset)
+  median_ts = get_adjusted_time(h);
+  if(b.timestamp < median_ts)
   {
-    timestamps.push_back(m_db->get_block_timestamp(offset));
+    MERROR_VER("Timestamp of block with id: " << get_block_hash(b) << ", " << b.timestamp << ", less than median of last " << BLOCKCHAIN_TIMESTAMP_CHECK_WINDOW << " blocks, " << median_ts);
+    return false;
   }
 
-  return check_block_timestamp(timestamps, b, median_ts);
```

Comment:
Is this 3-parameter overload of `check_block_timestamp()` being deprecated? If so, might be nice to remove it completely.

Reply:
It looks like it's still used.